src/Yarhl.UnitTests/IO/DataStreamTests.cs

```diff
@@ -965,6 +965,41 @@ public void ReadFormAfeterDisposeThrowException()
             Assert.Throws<ObjectDisposedException>(() => stream.ReadFormat<byte>());
         }
 
+        [Test]
+        public void SliceStream()
+        {
+            DataStream stream = new DataStream();
+            stream.WriteByte(0xBE);
+            stream.WriteByte(0xBA);
+            stream.WriteByte(0xCA);
+            stream.WriteByte(0xFE);
+
+            DataStream testSlice = stream.Slice(2);
+            Assert.That(
+                () => testSlice,
```

Comment:
This is impossible to throw excepcion right? Do you want to check if the constructor throw or if the variable is null?